src/shared/components/oncoprint/ResultsViewOncoprint.tsx

```diff
@@ -607,28 +625,48 @@ export default class ResultsViewOncoprint extends React.Component<IResultsViewOn
 
     @computed get heatmapTrackGroupsUrlParam() {
         return _.sortBy(this.molecularProfileIdToHeatmapTracks.values(), (x:HeatmapTrackGroupRecord)=>x.trackGroupIndex)
-            .filter((x:HeatmapTrackGroupRecord)=>!!x.genes.size)
-            .map((x:HeatmapTrackGroupRecord)=>`${x.molecularProfileId},${x.genes.keys().join(",")}`)
-            .join(";");
+        .filter((x:HeatmapTrackGroupRecord)=>!!x.entities.size)
+        .map((x:HeatmapTrackGroupRecord)=>`${x.molecularProfileId},${x.entities.keys().join(",")}`)
+        .join(";");
+    }
+
+    // treatments selected iin heatmap are added to the `treatment_list` url param
+    @computed get treatmentsUrlParam():string {
+        return _.filter(this.molecularProfileIdToHeatmapTracks.values(), (x:HeatmapTrackGroupRecord)=> x.molecularAlterationType === AlterationTypeConstants.TREATMENT_RESPONSE)
+        .map((x:HeatmapTrackGroupRecord)=>`${x.entities.keys().join(";")}`)
+        .join(";");
+    }
+
+    @computed get selectedHeatmapProfileAlterationType():string {
+        let molecularProfile = this.props.store.molecularProfileIdToMolecularProfile.result[this.selectedHeatmapProfile];
+        return molecularProfile.molecularAlterationType;
     }
 
-    private addHeatmapTracks(molecularProfileId:string, genes:string[]) {
-        let trackGroup = this.molecularProfileIdToHeatmapTracks.get(molecularProfileId);
-        if (!trackGroup) {
-            let newTrackGroupIndex = 2;
-            for (const group of this.molecularProfileIdToHeatmapTracks.values()) {
-                newTrackGroupIndex = Math.max(newTrackGroupIndex, group.trackGroupIndex + 1);
+    private addHeatmapTracks(molecularProfileId:string, entities:string[]) {
+        
+        let profile:MolecularProfile = this.props.store.molecularProfileIdToMolecularProfile.result![molecularProfileId];
+        // TODO fix the Promise dependency in this function (remove if profile statement)
```

Comment:
TODO?

Reply:
No, this was a leftover. I have removed the TODO comment.